src/BatchMailerMessage.php

```diff
@@ -283,7 +287,34 @@ public function headers(): array
     /** @return array<int, \Illuminate\Mail\Attachment> */
     public function attachments(): array
     {
-        return $this->attachments;
+        return array_merge($this->attachments, $this->rawAttachments);
+    }
+
+    public function getPreparedAttachments(): array
+    {
+        return $this->prepareAttachments();
+    }
+
+    private function prepareAttachments(): array
+    {
+        return collect([...$this->attachments, ...$this->rawAttachments])
+            ->map(static function (array|DataPart $attachment) {
+                if (is_array($attachment)) {
+                    return $attachment;
+                }
+
+                $filename = sprintf("batch-mailer-temp/%s.%s", time(), $attachment->getMediaSubtype());
```

Comment:
It's better to make this directory customizable to avoid conflicts in user applications